### PR TITLE
Fix replicate upto

### PR DESCRIPTION
**Pull request by cryptix**

### CODE REVIEW

**Comment by cblgh** [comment]
I'll take a look at this in the afternoon :3 

With this in place, @arj03 should see more reasonable numbers on the # of in-database messages when running against a go node in netsim, right? i.e. instead of the node saying it contains an _entire fixtures database_ worth of messages, it will report an accurate count (only the messages it cares about, as per its hops settings)

File: plugins/replicate/upto.go [hunk 1]
```diff
@@ -41,26 +51,33 @@ func (lt replicatePlug) Handler() muxrpc.Handler {
...
+	wantedSet.AddRef(g.self)
+	list, err := wantedSet.List()
 	if err != nil {
-		req.CloseWithError(fmt.Errorf("replicate: did not get feed source: %w", err))
-		return
+		return err
 	}
 
-	err = luigi.Pump(ctx, req.Stream, src)
+	set, err := ssb.WantedFeedsWithSequnce(g.users, list)
```
Comment (cblgh) [inline plugins/replicate/upto.go:67]: typo: 
```diff
-ssb.WantedFeedsWithSequnce
+-ssb.WantedFeedsWithSequence

Comment (cblgh) [reply plugins/replicate/upto.go:67]: (or just `ssb.WantedFeedsWithSeqs`)

**Review by cblgh — APPROVED** [review]
lgtm excepting the mentioned typo :~)

**Comment by cryptix** [comment]
> it will report an accurate count (only the messages it cares about, as per its hops settings)

yes, exactly. thanks for the summary.

**Comment by arj03** [comment]
I compiled go-ssb master and tried it out. Seems like I'm still seeing different numbers?

go:
```
ok 4 - load peer @TWKY4Bq5ezVqbXV2D7NyODxCXgu8o4rgp/sf1GdHbCw=.ed25519
ok 5 - enter server
ok 6 - hops server 1
ok 7 - load server @3eIPXptMcbrFPw8seKbywmPbRogERylNuoFVaZ9AlOg=.ed25519
ok 8 - alloffsets server
# waiting for server sbot to start; retry 0/15
ok 9 - start server go-sbot
# server (7867 messages) has id @3eIPXptMcbrFPw8seKbywmPbRogERylNuoFVaZ9AlOg=.ed25519 
# logging to server.txt
```

js:
```
ok 4 - load peer @TWKY4Bq5ezVqbXV2D7NyODxCXgu8o4rgp/sf1GdHbCw=.ed25519
ok 5 - skipoffset peer
ok 6 - enter server
ok 7 - hops server 1
ok 8 - load server @3eIPXptMcbrFPw8seKbywmPbRogERylNuoFVaZ9AlOg=.ed25519
ok 9 - alloffsets server
ok 10 - start server ssb-server
# server (6167 messages) has id @3eIPXptMcbrFPw8seKbywmPbRogERylNuoFVaZ9AlOg=.ed25519 
# logging to server.txt
```

script js:

```
comment booting

enter peer
hops peer 1
load peer @TWKY4Bq5ezVqbXV2D7NyODxCXgu8o4rgp/sf1GdHbCw=.ed25519
skipoffset peer

enter server
hops server 1
load server @3eIPXptMcbrFPw8seKbywmPbRogERylNuoFVaZ9AlOg=.ed25519
alloffsets server

start server ssb-server
start peer ssb-server

connect peer server
waituntil peer server@latest
stop peer
```

script go:


```
comment booting

enter peer
hops peer 1
load peer @TWKY4Bq5ezVqbXV2D7NyODxCXgu8o4rgp/sf1GdHbCw=.ed25519

enter server
hops server 1
load server @3eIPXptMcbrFPw8seKbywmPbRogERylNuoFVaZ9AlOg=.ed25519
alloffsets server

start server go-sbot
start peer ssb-server

connect peer server
waituntil peer peer@latest
stop peer

```

**Comment by arj03** [comment]
do I need changes in netsim as well?

**Comment by cryptix** [comment]
> do I need changes in netsim as well?

no you shouldn't. Which fixture are you using this with? the 9001 messages one?

**Comment by arj03** [comment]
yep with 20 authors

**Comment by cryptix** [comment]
@cblgh had a good intuition and remembered something I mentioned just in passing.

go-sbot has a slightly different [defintion](https://github.com/cryptoscope/ssb/blob/2526c35609f627c60bbe32b2a226fec6cb603379/graph/builder.go#L319) of hops:0 and hops:1. Which translates to go hops = node hops + 1.

I changed the go-ssb shim to use `  -hops "$(( ${HOPS} - 1 ))" ` and get matching numbers.

**Comment by arj03** [comment]
Can confirm that behaviour

**Pull request merged**